test/connectivity/connectivityTest.sh

```diff
@@ -0,0 +1,258 @@
+#!/bin/bash
+
+###############################################################################
+# This script is used to run connectivity test for Linux.
+###############################################################################
+set -e
+
+# Import test-related functions
+. $(dirname "$0")/testHelper.sh
+
+function prepare_test_from_artifacts() {
+    print_highlighted_message 'Prepare test from artifacts'
+
+    echo 'Remove working folder'
+    rm -rf "$working_folder"
+    mkdir -p "$working_folder"
+
+    declare -a pkg_list=( $iotedged_artifact_folder/*.deb )
+    iotedge_package="${pkg_list[*]}"
+    echo "iotedge_package=$iotedge_package"
+
+    echo 'Extract quickstart to working folder'
+    mkdir -p "$quickstart_working_folder"
+    tar -C "$quickstart_working_folder" -xzf "$iotedge_quickstart_artifact_file"
+
+    echo "Copy deployment file from $connectivity_deployment_artifact_file"
+    cp "$connectivity_deployment_artifact_file" "$deployment_working_file"
+
+    sed -i -e "s@<Analyzer.ConsumerGroupId>@$EVENT_HUB_CONSUMER_GROUP_ID@g" "$deployment_working_file"
+    sed -i -e "s@<Analyzer.EventHubConnectionString>@$EVENTHUB_CONNECTION_STRING@g" "$deployment_working_file"
+    sed -i -e "s@<Architecture>@$image_architecture_label@g" "$deployment_working_file"
+    sed -i -e "s/<Build.BuildNumber>/$ARTIFACT_IMAGE_BUILD_NUMBER/g" "$deployment_working_file"
+    sed -i -e "s@<Container_Registry>@$CONTAINER_REGISTRY@g" "$deployment_working_file"
+    sed -i -e "s@<CR.Username>@$CONTAINER_REGISTRY_USERNAME@g" "$deployment_working_file"
+    sed -i -e "s@<CR.Password>@$CONTAINER_REGISTRY_PASSWORD@g" "$deployment_working_file"
+
+    sed -i -e "s@<LoadGen.MessageFrequency>@$LOADGEN_MESSAGE_FREQUENCY@g" "$deployment_working_file"
+}
+
+function print_logs() {
+    local ret=$1
+    local test_end_time=$2
+    local elapsed_seconds=$3
+
+    elapsed_time="$(TZ=UTC0 printf '%(%H:%M:%S)T\n' "$elapsed_seconds")"
+    print_highlighted_message "Test completed at $test_end_time, took $elapsed_time."
+
+    if (( ret < 1 )); then
+        return;
+    fi
+
+    print_highlighted_message 'Print logs'
+    print_highlighted_message 'LOGS FROM IOTEDGED'
+    journalctl -u iotedge -u docker --since "$test_start_time" --no-pager || true
```

Comment:
instead of just return `true` maybe we should also print out that the logging failed?

Reply:
improve in other PR.